benchmarks/helics/helics_benchmark_main.h

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright (c) 2017-2019,
+Battelle Memorial Institute; Lawrence Livermore National Security, LLC; Alliance for Sustainable Energy, LLC.  See
+the top-level NOTICE for additional details. All rights reserved.
+SPDX-License-Identifier: BSD-3-Clause
+*/
+#pragma once
+#include <benchmark/benchmark.h>
+#include "helics/helics-config.h"
+#include <iostream>
+#ifdef ENABLE_ZMQ_CORE
+#include "helics/core/zmq/ZmqCommsCommon.h" 
+#endif
+
+// Helper macro to create a main routine in a test that runs the benchmarks
+#define HELICS_BENCHMARK_MAIN(label)                                                                              \
+    int main (int argc, char **argv)                                                                              \
+    {                                                                                                             \
+        ::benchmark::Initialize (&argc, argv);                                                                    \
+        if (::benchmark::ReportUnrecognizedArguments (argc, argv))                                                \
+            return 1;                                                                                             \
+        std::cout << "#benchmark:" << #label << '\n';                                                             \
```

Comment:
Do the #'s do anything special in macros?

Reply:
The first one is just to indicate a comment that we might want to trigger text processing on.  so that isn't special.  The second one on`#label` is a command to stringify the argument, so to treat it as a string rather than try to evaluate it in the `C++` context.  Basically means the preprocessor will put whatever you type in the macro argument as a string.   so if you did something like argument*7+4   it would make a string "argument*7+4"  instead of trying to evaluate it and then feed to the output stream.